Mathlib/MeasureTheory/Integral/RieszMarkovKakutani.lean

```diff
@@ -105,3 +125,135 @@ theorem rieszContentAux_sup_le (K1 K2 : Compacts X) :
   rw [add_assoc, add_comm (ε / 2), add_assoc, add_halves ε, add_assoc]
 
 end RieszSubadditive
+
+
+section PartitionOfUnity
+
+variable [T2Space X] [LocallyCompactSpace X]
+
+lemma exists_sum_one_of_isCompact_nnreal
+    {s₀ s₁ : Set X} {t : Set X} (s₀_compact : IsCompact s₀) (s₁_compact : IsCompact s₁)
+    (t_compact : IsCompact t) (disj : Disjoint s₀ s₁) (hst : s₀ ∪ s₁ ⊆ t) :
+    ∃ (f₀ f₁ : C_c(X, ℝ≥0)), EqOn f₀ 1 s₀ ∧ EqOn f₁ 1 s₁ ∧ EqOn (f₀ + f₁) 1 t := by
+  set so : Fin 2 → Set X := fun j => if j = 0 then s₀ᶜ else s₁ᶜ with hso
+  have soopen : ∀ j, IsOpen (so j) := by
+    intro j
+    by_cases h0 : j = 0
+    · rw [h0, hso]
+      simp only [Fin.isValue, ↓reduceIte, isOpen_compl_iff]
+      exact IsCompact.isClosed <| s₀_compact
+    · rw [hso]
+      simp only [Fin.isValue]
+      rw [if_neg h0]
+      simp only [Fin.isValue, isOpen_compl_iff]
+      exact IsCompact.isClosed <| s₁_compact
```

Comment:
```suggestion
  have soopen (j : Fin 2) :  IsOpen (so j) := by
    fin_cases j
    · simp only [hso, Fin.zero_eta, Fin.isValue, ↓reduceIte, isOpen_compl_iff]
      exact IsCompact.isClosed <| s₀_compact
    · simp only [hso, Fin.isValue, Fin.mk_one, one_ne_zero, ↓reduceIte, isOpen_compl_iff]
      exact IsCompact.isClosed <| s₁_compact
```
Just to learn a few tricks. Often, `rw` and `simp` in line can be compressed into a single `simp` call.